src/libraries/System.Runtime/ref/System.Runtime.cs

```diff
@@ -15158,7 +15158,6 @@ public enum LazyThreadSafetyMode
         PublicationOnly = 1,
         ExecutionAndPublication = 2,
     }
-    [System.Runtime.Versioning.RequiresPreviewFeaturesAttribute]
```

Comment:
Are we ready to start using Lock in runtime? Would the general philosophy be to use it anywhere we're currently using a dedicated lock object and it's only being used for enter/exit and not signaling?

Reply:
yeah we were planning to initially start using it in Threadpool synchronization logic. But might be good to start using in few other places too, so we can iron out issues (if any) before going broader.

Reply:
Yea we were planning on using it in a few places relevant to the thread pool to start with. Those locks are used frequently but aren't very contentious. It is already used in NativeAOT, wanted to get some more coverage in CoreCLR as well. It should be ready to be used anywhere Monitor is used currently just as a lock. There are some SOS diagnostics to add, but it should be fine to update existing uses.

Reply:
Perf wise, how should I think about it? It'll be as-good-or-better in pretty much any scenario, such that I can use it without doing significant perf validation, or it's more nuanced?